sdk/openai/openai/test/public/node/whisper.spec.ts

```diff
@@ -61,4 +53,45 @@ describe("OpenAI", function () {
       );
     });
   });
+
+  describe("Memory Usage", function () {
+    let recorder: Recorder;
+    let client: OpenAIClient;
+    const authMethod: AuthMethod = "OpenAIKey";
+    beforeEach(async function (this: Context) {
+      recorder = await startRecorder(this.currentTest);
+      client = createClient(authMethod, { recorder });
+    });
+
+    afterEach(async function () {
+      if (recorder) {
+        await recorder.stop();
+      }
+    });
+
+    it("stream 10MB file without loading it", async function () {
```

Comment:
Is this specific to code in openai, or are you really just testing that core-rest-pipeline flows data from a stream without buffering it all into memory?

Either way, I think you could also verify this with a unit test, that might be simpler and more direct than checking memory during a live test (though you may want both for full coverage):

1. Create a client with a mock http transport
2. Call `getAudioTranscription()` with a stream that returns a small amount of data (enough to fill all the internal buffers) then blocks.  1MB should be plenty, but you might be able to use even a very small amount like 4k, 8k, etc.
3. Verify the client has sent data to the mock http transport while the input stream is blocked.  This verifies the client is flowing data from the input stream to the http transport without buffering too much in memory.

CC: @johanste who may know other ways to test this.

Reply:
core-rest-pipeline can handle both stream and non-stream cases and the intent here is to test that this library creates a streamable file object before it is sent to core-rest-pipeline, see:
https://github.com/Azure/azure-sdk-for-js/blob/81fdbcb7f49cbf35cdb703f5ae8ef09133fcdea3/sdk/openai/openai/src/api/operations.ts#L710

I'll play with your unit test idea, thanks for the suggestion!

Reply:
I added a unit test in https://github.com/Azure/azure-sdk-for-js/pull/27743/commits/9f42ba18a436db0d3a34c07f9abd7538d66c24e4.